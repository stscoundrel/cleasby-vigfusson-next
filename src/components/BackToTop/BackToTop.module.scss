.backToTop {
  position: fixed;
  bottom: 15px;
  right: 15px;
  width: 36px;
  height: 36px;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: $color-blue;
  color: white;
  border-radius: 50%;
  border: 1px solid white;
  font-weight: bold;
  cursor: pointer;
}

@media only screen and (min-width: 768px) {
  .backToTop {
    bottom: 30px;
    right: 30px;
  }
}