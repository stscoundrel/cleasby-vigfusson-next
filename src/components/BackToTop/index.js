import styles from './BackToTop.module.scss'

export default function BackToTop() {
  const scrollToTop = () => {
    window.scrollTo({
      top: 0,
      behavior: 'smooth',
    })
  }

  return (
    <div className={styles.backToTop} aria-label="Back to top" role="button" onClick={() => scrollToTop()}>
      â†‘
    </div>
  )
}
