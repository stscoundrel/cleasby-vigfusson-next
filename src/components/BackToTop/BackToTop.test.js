import React from 'react'
import renderer from 'react-test-renderer'
import BackToTop from './index'

describe('BackToTop button', () => {
  test('Matches the snapshot', () => {
    const tree = renderer.create(<BackToTop />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Triggers scroll after click', async () => {
    const mockScroll = jest.fn()
    window.scrollTo = mockScroll

    let tree

    await renderer.act(async () => {
      tree = renderer.create(<BackToTop />)
    })

    await renderer.act(async () => {
      await tree.root.findByType('div').props.onClick()
    })

    expect(mockScroll).toHaveBeenCalled()
  })
})
