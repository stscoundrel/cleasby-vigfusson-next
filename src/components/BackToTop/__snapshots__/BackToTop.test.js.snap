// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackToTop button Matches the snapshot 1`] = `
<div
  aria-label="Back to top"
  className="backToTop"
  onClick={[Function]}
  role="button"
>
  â†‘
</div>
`;
